src/useMercureSubscription.js

```diff
@@ -0,0 +1,45 @@
+import {
+  CRUD_GET_ONE_SUCCESS,
+  FETCH_END,
+  GET_ONE,
+  useDataProvider,
+} from 'ra-core';
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+
+export default function useMercureSubscription(resource, idOrIds) {
+  const dataProvider = useDataProvider();
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const ids = Array.isArray(idOrIds) ? idOrIds : [idOrIds];
+
+    if (
+      dataProvider.subscribe === undefined ||
+      dataProvider.unsubscribe === undefined
+    ) {
+      console.warn(
```

Comment:
This console message will probably show a lot. Maybe we should use a ref to show it only once?